app/controllers/declaration/AdditionalDeclarationTypeController.scala

```diff
@@ -48,15 +51,21 @@ class AdditionalDeclarationTypeController @Inject()(
   }
 
   def submitForm(): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
-    val decType = extractFormType(request)
+    val action = FormAction.bindFromRequest
+    val decType = extractFormType(request).form().bindFromRequest()
+
     decType
-      .form()
-      .bindFromRequest()
       .fold(
         formWithErrors => Future.successful(BadRequest(declarationTypePage(formWithErrors))),
         validAdditionalDeclarationType =>
-          updateCache(validAdditionalDeclarationType)
-            .map(_ => Redirect(controllers.declaration.routes.ConsignmentReferencesController.displayPage()))
+          updateCache(validAdditionalDeclarationType).map { _ =>
+            action match {
+              case Some(SaveAndContinue) =>
+                Redirect(controllers.declaration.routes.ConsignmentReferencesController.displayPage())
+              case Some(SaveAndReturn) =>
+                navigator.goToDraftConfirmation()
```

Comment:
So the only way to "save and resume" is to complete the page with valid data (agreed with Ian & Ben M)

Reply:
Is SaveAndReturn always goes to this page ?

Reply:
Yep

Reply:
Perhaps we can expand the navigator eventually to redirect based on the action